### PR TITLE
Feature/snowflake e2e test

### PR DESCRIPTION
**Pull request by zhangchi1**
**What this PR does / why we need it**:
1. Added integration tests for snowflake batch store in serving 

**Next Step**
1. Mocking snowflake database 

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: serving/src/test/java/feast/serving/it/ServingServiceSnowflakeStorageIT.java [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+  public KafkaTemplate<String, FeatureRow> specKafkaTemplate() {
+
+    Map<String, Object> props = new HashMap<>();
+    props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "kafka:9092,localhost:9094");
+
+    KafkaTemplate<String, FeatureRow> t =
+        new KafkaTemplate<>(
+            new DefaultKafkaProducerFactory<>(
+                props, new StringSerializer(), new KafkaSerialization.ProtoSerializer<>()));
+    t.setDefaultTopic("feast-features");
```
Comment (swatiarora0208) [inline serving/src/test/java/feast/serving/it/ServingServiceSnowflakeStorageIT.java]: topic is used as multiple place .. could be defined as constant

File: storage/connectors/jdbc/pom.xml [hunk 2]
```diff
@@ -146,6 +162,12 @@
...
         </dependency>
+<!--        For testing DB only -->
+        <dependency>
+            <groupId>org.xerial</groupId>
+            <artifactId>sqlite-jdbc</artifactId>
+            <version>3.16.1</version>
+        </dependency>
     </dependencies>
     <properties>
         <maven.test.skip>true</maven.test.skip>
```
Comment (zhangchi1) [inline storage/connectors/jdbc/pom.xml]: I think we should un-skip the new mocking tests. 
And skip those manual tests, assume we still want to include those manual e2e tests in our snowflake branch.

File: serving/src/test/java/feast/serving/it/ServingServiceSnowflakeStorageIT.java (hunk 1 above)
Comment (zhangchi1) [reply serving/src/test/java/feast/serving/it/ServingServiceSnowflakeStorageIT.java]: Done

File: infra/docker/core/Dockerfile.dev [hunk 3]
```diff
@@ -1,6 +1,6 @@
 FROM openjdk:11-jre
 ARG REVISION=dev
-ADD $PWD/core/target/feast-core-$REVISION.jar /opt/feast/feast-core.jar
+ADD $PWD/core/target/feast-core-$REVISION-exec.jar /opt/feast/feast-core.jar
```
Comment (jmelinav) [inline infra/docker/core/Dockerfile.dev]: this should already be there in master, I pushed it last week.

File: infra/docker/serving/Dockerfile.dev [hunk 4]
```diff
@@ -7,7 +7,7 @@ ARG REVISION=dev
 RUN wget -q https://github.com/grpc-ecosystem/grpc-health-probe/releases/download/v0.3.1/grpc_health_probe-linux-amd64 \
          -O /usr/bin/grpc-health-probe && \
     chmod +x /usr/bin/grpc-health-probe
-ADD $PWD/serving/target/feast-serving-$REVISION.jar /opt/feast/feast-serving.jar
+ADD $PWD/serving/target/feast-serving-$REVISION-exec.jar /opt/feast/feast-serving.jar
```
Comment (jmelinav) [inline infra/docker/serving/Dockerfile.dev]: same as above.

File: serving/src/test/java/feast/serving/it/ServingServiceSnowflakeStorageIT.java [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.containers.DockerComposeContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@ActiveProfiles("it")
+@SpringBootTest()
+@Testcontainers
+public class ServingServiceSnowflakeStorageIT {
```
Comment (zhangchi1) [inline serving/src/test/java/feast/serving/it/ServingServiceSnowflakeStorageIT.java:61]: Disable this manual Integration test

File: serving/pom.xml [hunk 6]
```diff
@@ -340,6 +346,11 @@
       <version>0.4.4-alpha.1</version>
       <scope>test</scope>
     </dependency>
+     <!-- Kafka -->
+    <dependency>
```
Comment (jmelinav) [inline serving/pom.xml:366]: if it is used only for test add the scope as test.

File: storage/connectors/jdbc/src/test/java/feast/storage/connectors/jdbc/retriever/SnowflakeHistoricalRetrieverJSONColTest.java [hunk 7]
```diff
@@ -40,7 +40,7 @@
   private String staging_location = "s3://feast-snowflake-staging/test/";
 
   private Map<String, String> snowflakeConfig = new HashMap<>();
-  private String SFUrl = "jdbc:snowflake://ry42518.us-east-2.aws.snowflakecomputing.com";
+  private String SFUrl = "jdbc:snowflake://nx46274.us-east-2.aws.snowflakecomputing.com";
```
Comment (zhangchi1) [inline storage/connectors/jdbc/src/test/java/feast/storage/connectors/jdbc/retriever/SnowflakeHistoricalRetrieverJSONColTest.java]: remove all personal "real" url variables

File: serving/src/test/java/feast/serving/it/SnowflakeTestUtils.java [hunk 8]
```diff
@@ -0,0 +1,299 @@
...
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Scanner;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.zip.GZIPInputStream;
+import org.apache.commons.lang3.tuple.Pair;
+
+public class SnowflakeTestUtils {
```
Comment (jmelinav) [inline serving/src/test/java/feast/serving/it/SnowflakeTestUtils.java]: merge it with AuthTestUtils.java and rename file.

File: serving/src/main/resources/application.yml [hunk 9]
```diff
@@ -78,6 +78,24 @@ feast:
...
+    - name: historical_snowflake
+      type: JDBC
+      config:  # Store specific configuration.
+        schema: PUBLIC
+        database: DEMO_DB
+        role: ACCOUNTADMIN
+        table_name: FEAST_FEATURES_TEST
+        url: jdbc:snowflake://test.snowflakecomputing.com
+        username: username
+        password: password
```
Comment (zhangchi1) [inline serving/src/main/resources/application.yml:90]: name those configs as fake-configs

File: serving/src/test/resources/application-it.yml [hunk 10]
```diff
@@ -0,0 +1,26 @@
+feast:
+
+  # Indicates the active store. Only a single store in the last can be active at one time. In the future this key
+  # will be deprecated in order to allow multiple stores to be served from a single serving instance
```
Comment (jmelinav) [inline serving/src/test/resources/application-it.yml]: there is already application-it.properties. put the defaults there.

File: serving/src/test/resources/docker-compose/docker-compose-it-core.yml [hunk 11]
```diff
@@ -2,8 +2,8 @@ version: '3'
 
 services:
   core:
-    image: gcr.io/kf-feast/feast-core:latest
-    volumes: 
+    image: feast-core:dev1
```
Comment (jmelinav) [inline serving/src/test/resources/docker-compose/docker-compose-it-core.yml]: undo this.

File: serving/src/test/java/feast/serving/it/ServingServiceSnowflakeStorageIT.java (hunk 5 above)
Comment (zhangchi1) [inline serving/src/test/java/feast/serving/it/ServingServiceSnowflakeStorageIT.java:61]: Add a comment for using customize Feast Core

**Pull request merged**